### PR TITLE
feat: chocolatey package scripts

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Signed-off-by: Darren Murray <darren.murray@lacework.net>

## Summary

Create a powershell script to update and publish the Lacework CLI as a chocolatey package

## How did you test this change?

New Codefresh Pipeline

## Issue

https://lacework.atlassian.net/browse/ALLY-76

### CODE REVIEW

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Review by afiune — COMMENTED** [review]
Can we add honeycomb fields just like we did for homebrew?

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco push

**Comment by dmurray-lacework** [comment]
choco push

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

**Comment by dmurray-lacework** [comment]
choco pack

File: cli/cmd/honeyvent.go [hunk 1]
```diff
@@ -50,6 +50,10 @@ const (
 	// install method was via homebrew package manager
 	HomebrewInstall = "LW_HOMEBREW_INSTALL"
 
+	// ChocolateyInstall is an environment variable that denotes the
+	// install method was via chocolatey package manager
+	ChocolateyInstall = "LW_CHOCOLATEY_INSTALL"
```
Comment (afiune) [inline cli/cmd/honeyvent.go:55]: How are we setting this env var?

Comment (dmurray-lacework) [reply cli/cmd/honeyvent.go:55]: in the ChocolateyInstall.ps1

File: cli/cmd/cli_windows.go [hunk 2]
```diff
@@ -31,6 +35,12 @@ var promptIconsFunc = func(icons *survey.IconSet) {
 // UpdateCommand returns the command that a user should run to update the cli
 // to the latest available version (windows specific command)
 func (c *cliState) UpdateCommand() string {
+	if os.Getenv(ChocolateyInstall) != "" {
+		return `
+  choco upgrade lacework-cli
+`
```
Comment (rmoles) [inline cli/cmd/cli_windows.go:41]: nit: IMO this would be nicer as a single line

File: scripts/chocolatey/build/tools/chocolateyinstall.ps1 [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+.DESCRIPTION
+This script installs the Lacework Command Line Interface.
+
+#>
+
+$ErrorActionPreference = "stop"
+
+Set-Variable GithubReleasesRootUrl -Option ReadOnly -value "https://github.com/lacework/go-sdk/releases" -Force
+Set-Variable PackageName -Option ReadOnly -value "lacework-cli-windows-amd64.exe.zip" -Force
```
Comment (rmoles) [inline scripts/chocolatey/build/tools/chocolateyinstall.ps1:15]: I notice this is assuming `amd64` do we need to add any consideration for any different arch types?

File: scripts/chocolatey/build/tools/chocolateyinstall.ps1 [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    $result = @{ "zip" = $cli_dest }
+
+    try
+    {
+        Get-File $sha_url $sha_dest
+        $result["shasum"] = (Get-Content $sha_dest).Split()[0]
+    }
+    catch
+    {
+        Write-Warning "No shasum exists for $version. Skipping validation."
```
Comment (rmoles) [inline scripts/chocolatey/build/tools/chocolateyinstall.ps1:60]: Wondering if we should throw an error here? Is there a valid scenario where the CLI wouldn't have a shasum?

File: scripts/chocolatey/build/tools/chocolateyinstall.ps1 (hunk 3 above)
Comment (dmurray-lacework) [reply scripts/chocolatey/build/tools/chocolateyinstall.ps1:15]: Our existing installer doesn't

File: scripts/chocolatey/build/tools/chocolateyinstall.ps1 (hunk 4 above)
Comment (dmurray-lacework) [reply scripts/chocolatey/build/tools/chocolateyinstall.ps1:60]: That's a good point, let me raise a ticket to check this on the existing installer as well

**Pull request merged**